File: AssetManager/Data/DataComms/AssetManager/DBParameter.vb
Public Class DBParameter
    Public Property FieldName As String
    Public Property Value As Object

    Sub New(ByVal fieldName As String, ByVal fieldValue As Object)
        Me.FieldName = fieldName
        Me.Value = fieldValue
    End Sub

End Class

Public Class DBQueryParameter
    Inherits DBParameter
    Public Property IsExact As Boolean
    Public Property OperatorString As String

    Sub New(ByVal fieldName As String, ByVal fieldValue As Object, operatorString As String)
        MyBase.New(fieldName, fieldValue)
        Me.IsExact = IsExact
        Me.OperatorString = operatorString
    End Sub

    Public Sub New(ByVal fieldName As String, ByVal fieldValue As Object, isExact As Boolean)
        MyBase.New(fieldName, fieldValue)
        Me.IsExact = isExact
        Me.OperatorString = "AND"
    End Sub

    Public Sub New(ByVal fieldName As String, ByVal fieldValue As Object, isExact As Boolean, operatorString As String)
        MyBase.New(fieldName, fieldValue)
        Me.IsExact = isExact
        Me.OperatorString = operatorString
    End Sub
End Class

File: AssetManager/Data/DataComms/AssetManager/DBParameter.cs
namespace AssetManager
{
    public class DBParameter
    {
        public string FieldName { get; set; }
        public object Value { get; set; }

        public DBParameter(string fieldName, object fieldValue)
        {
            this.FieldName = fieldName;
            this.Value = fieldValue;
        }

    }
}
namespace AssetManager
{

    public class DBQueryParameter : DBParameter
    {
        public bool IsExact { get; set; }
        public string OperatorString { get; set; }

        public DBQueryParameter(string fieldName, object fieldValue, string operatorString) : base(fieldName, fieldValue)
        {
            this.IsExact = IsExact;
            this.OperatorString = operatorString;
        }

        public DBQueryParameter(string fieldName, object fieldValue, bool isExact) : base(fieldName, fieldValue)
        {
            this.IsExact = isExact;
            this.OperatorString = "AND";
        }

        public DBQueryParameter(string fieldName, object fieldValue, bool isExact, string operatorString) : base(fieldName, fieldValue)
        {
            this.IsExact = isExact;
            this.OperatorString = operatorString;
        }

    }
}
